### ISSUE TITLE
Use separate kustomize binary if available (keep kubectl as fallback)

### ISSUE THREAD

**Issue by LukasGentele** [devspace-cloud/devspace#859]
**Is your feature request related to a problem?**  
Yes. From Slack: "I had to build a shell script in order to call kustomize directly, as we’re using a newer version with alpha plugins enabled (another flag)"

**Which solution do you suggest?**  
Check for kustomize binary in PATH and prefer to use it over kubectl.

**Which alternative solutions exist?**  
Adding some kind of config flag?

**Additional context**  

Credits to @korbin for suggesting this via Slack.

/kind feature

**Comment by Inlustra** [comment]
Bumping this as I'm facing the same problem.
According to the docs, this should automatically happen:
https://devspace.sh/cli/docs/configuration/deployments/kustomizations

But having a quick search through the code, I'm pretty sure it just uses the --kustomize/-k flag...
https://github.com/devspace-cloud/devspace/blob/master/pkg/devspace/deploy/deployer/kubectl/kubectl.go

**Comment by FabianKramm** [comment]
@Inlustra yes currently we only use the `--kustomize/-k` flag with kubectl. I'll implement a more flexible version with kustomize on friday though (release either on friday or somewhere in the beginning of next week), since this is requested a lot recently

**Comment by Inlustra** [comment]
@FabianKramm Sorry to bring this up, I've just had time to run and give this a go! 
Unfortunately I still can't seem to get it to work, I'm installing them like so:
```sh
          KUSTOMIZE_VERSION=3.5.4
          DEVSPACE_VERSION=4.12.1
          curl -L https://github.com/kubernetes-sigs/kustomize/releases/download/kustomize%2Fv${KUSTOMIZE_VERSION}/kustomize_v${KUSTOMIZE_VERSION}_linux_amd64.tar.gz | tar -xzv
          chmod +x kustomize
          sudo mv kustomize /usr/local/bin
          kustomize version
          curl -s -L "https://github.com/devspace-cloud/devspace/releases/tag/v${DEVSPACE_VERSION}" | sed -nE 's!.*"([^"]*devspace-linux-amd64)".*!https://github.com\1!p' | xargs -n 1 curl -L -o devspace && chmod +x devspace;
          sudo mv devspace /usr/local/bin
          devspace --version
```
Which is working perfectly, Running `kustomize build k8s/overlays/prod` immediately after is also successful.
However: `devspace deploy -p production --wait --timeout=300` fails with 
```
error: json: unknown field "replicas"
``` 
As it is still using `kubectl -k`. Am I missing something? 
My Devspace.yml: 
```
version: v1beta7
images:
  dt_backend:
    image: dt_backend
    tags:
      - ${dt_backend}-${DEVSPACE_RANDOM}
    dockerfile: ./k8s/backend.Dockerfile
    cmd: ["yarn", "watch"]
  dt_frontend:
    image: dt_frontend
    tags:
      - ${dt_frontend}-${DEVSPACE_RANDOM}
    dockerfile: ./k8s/frontend.Dockerfile
    cmd: ["yarn", "watch"]
deployments:
  - name: dt
    kubectl:
      kustomize: true
      replaceImageTags: false
      manifests:
        - k8s/overlays/local
dev:
  autoReload:
    paths:
      - ./package.json
      - "packages/**/package.json"
  open:
    - url: http://localhost:4000
  sync:
    - imageName: dt_backend
      waitInitialSync: true
      localSubPath: ./
      containerPath: /opt/app
      uploadExcludePaths:
        - packages/schema/dist
        - "**/node_modules"
      excludePaths:
        - "**/.DS_Store"
        - "**/*.log"
        - "**/Dockerfile"
        - "**/Dockerfile.*"
        - "**/.env"
        - scripts/
        - devspace.yaml
    - imageName: dt_frontend
      waitInitialSync: true
      localSubPath: ./
      containerPath: /opt/app
      uploadExcludePaths:
        - packages/schema/dist
        - "**/node_modules"
      excludePaths:
        - "**/.DS_Store"
        - "**/*.log"
        - "**/Dockerfile"
        - "**/Dockerfile.*"
        - "**/.env"
        - scripts/
        - devspace.yaml

profiles:
  - name: production
    patches:
      # Remove the dev dockerfile command
      - op: remove
        path: images.dt_backend.cmd
      - op: remove
        path: images.dt_frontend.cmd
        # Use prod kustomize overlay
      - op: replace
        path: deployments.name=dt.kubectl.manifests
        value: ["k8s/overlays/prod"]
      - op: replace
        path: deployments.name=dt.kubectl.replaceImageTags
        value: true

vars:
  - name: dt_frontend
    default: "local"
  - name: dt_backend
    default: "local"
  - name: dt_schema
    default: "local"
  - name: dt_vscode
    default: "local"
```

**Comment by FabianKramm** [comment]
@Inlustra thanks for reporting! Seems like kustomize changed their `kustomize version` output which is why devspace assumes there is a wrong kustomize binary installed, so it currently works only with kustomize up to `v3.3ish`. I'll fix this for the next release!

**Comment by Inlustra** [comment]
Amazing! Thanks @FabianKramm, can't wait to use the new Kustomize features! 😁

**Comment by FabianKramm** [comment]
@Inlustra should be fixed now in v4.12.2
